test/extensions/wasm/test_data/stats_cpp.cc

```diff
@@ -44,7 +44,8 @@ extern "C" EMSCRIPTEN_KEEPALIVE void proxy_onTick() {
   logInfo(std::string("get counter = ") + std::to_string(c->get("test_tag")));
   logWarn(std::string("get gauge = ") + std::to_string(g->get(9)));
 
-  auto h_id = h->resolve(7, "test_tag", true);
```

Comment:
For the test did you intend h to be of type histogram ?

From an api perspective partiallyReolve should return a new counter, instead of mutating the current one.

Reply:
Yes!  Thanx, fixed.

Reply:
BTW: I changed partiallyResolve as per your suggestion.